libecole/src/reward/dualintegral.cpp

```diff
@@ -0,0 +1,51 @@
+#include "ecole/reward/dualintegral.hpp"
+#include "ecole/scip/model.hpp"
+#include "ecole/utility/chrono.hpp"
+
+namespace ecole::reward {
+
+namespace{
+  static auto dual_bound(scip::Model& model) {
+	switch (model.get_stage()) {
+	// Only stages when the following call is authorized
+		case SCIP_STAGE_TRANSFORMED:
+		case SCIP_STAGE_INITPRESOLVE:
+		case SCIP_STAGE_PRESOLVING:
+		case SCIP_STAGE_EXITPRESOLVE:
+		case SCIP_STAGE_PRESOLVED:
+		case SCIP_STAGE_INITSOLVE:
+		case SCIP_STAGE_SOLVING:
+		case SCIP_STAGE_SOLVED:
+			  return SCIPgetDualbound(model.get_scip_ptr());
+		default:
+			  return decltype(SCIPgetDualbound(nullptr)){0};
```

Comment:
We should let the users decide which value they want to use to compute the integral when there is no dual bound yet. A default at zero is ok maybe, but we should offer the possibility for users to set this value in the constructor as well.